src/inmanta/server/protocol.py

```diff
@@ -462,7 +462,7 @@ def abort(self):
 
 
 class SessionListener(object):
-    def new_session(self, session: Session) -> None:
+    async def new_session(self, session: Session) -> None:
```

Comment:
This is scary for performance reasons. It queues up all incoming agent calls on the session lock. 
(but it could still make sense, could you document a bit more)

Reply:
(but then again, these are longpoll sessions, so the call doesn't return either way)

Reply:
The only reason why this was changed, is to be able to catch exceptions throws by the `_register_session()` method in the agentmanager when the database connection is lost. Otherwise the `get_or_create_session()` in inmanta.server.protocol would grant a session while no primary could be elected in the agentmanager. Does that make sense?

Reply:
yes.